File: 0012-integer-to-roman/0012-integer-to-roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        vector<pair<int, string>> values{
            {1, "I"}, {4, "IV"}, {5, "V"},\
            {9, "IX"}, {10, "X"}, {40, "XL"},\
            {50, "L"}, {90, "XC"}, {100, "C"},\
            {400, "CD"}, {500, "D"}, {900, "CM"},\
            {1000, "M"}
        };
        
        int n = values.size() - 1;
        string ans = "";
        
        while (num > 0){
            for (int i = n; i >= 0; --i){
                auto [val, roman] = values[i];
                while (num >= val){
                    ans += roman;
                    num -= val;
                }
            }
        }
        
        return ans;
    }
};

File: 0012-integer-to-roman/0012-integer-to-roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        digits = [
            (1000, "M"), (900, "CM"), (500, "D"),\
            (400, "CD"), (100, "C"), (90, "XC"),\
            (50, "L"), (40, "XL"), (10, "X"),\
            (9, "IX"), (5, "V"), (4, "IV"), (1, "I")
        ]
        
        roman_digits = []
        
        # Loop through each symbol.
        for val, symbol in digits:
            # We don't want to continue looping if we're done
            if num == 0:
                break
            count, num = divmod(num, val)
            # Append "count" copies of "symbol" to roman_digits.
            roman_digits.append(symbol * count)
        
        return "".join(roman_digits)